### ISSUE TITLE
[CLOSED] Minimal pair task, maskLM acceptability, relocate NPI task

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#544]
<a href="https://github.com/HaokunLiu"><img src="https://avatars0.githubusercontent.com/u/35565210?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [HaokunLiu](https://github.com/HaokunLiu)**
_Friday Apr 12, 2019 at 21:06 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/pull/544_

----

Introducing minimal pair task, its input is a pair of sentences, but it does not inherit from pair classification task, otherwise bert model will use "[CLS] sent1 [SEP] sent2". Two task names are registered, to differentiate between frozen and tuned setting. 

"cola-pair-tuned" compute the acceptability logits on two sentences independently with the classifier 
 trained on cola and choose the more likely one. "cola-pair-frozen" predict the word at the location where two sentences differs with pretrained MaskLM head.

Borrowing classifier from another task was initially introduced only for mnli-diagnostic, therefore the control logic in main.py are written specifically for mnli-diagnostic. This update generalize that mechanism and allow other tasks to borrow classifier from a intrinsically similar task.
To use this feature, set TASK1 {use_classifier = "TASK0"} in default.conf, so that TASK1 will reuse the classifier of TASK0, instead of create its own.

----

_**[HaokunLiu](https://github.com/HaokunLiu)** included the following code: https://github.com/nyu-mll/jiant/pull/544/commits_
